### ISSUE TITLE
[Bug] [Crash] Open an incomplete link (http / https / ftp)

### ISSUE THREAD

**Issue by stodar** [mozilla-mobile/fenix#2585]
## Steps to reproduce
Enter an incomplete link in the address bar
Example: "ftp://127" or "http://exemple"

### Expected behavior
The browser must tell us that the link is wrong or redirected to a search

### Actual behavior
The browser crash and you have to clear the cache to access it again.
Poc: https://drive.google.com/open?id=1KCrCNzO0pwXEgwAsxhz3laibrcLef49e

### Device information
* Android device: Redmi note 5
* Android version: 8.1.0
* Fenix version: 1.0.1920

**Comment by kbrosnan** [comment]
I can't reproduce using

* Nexus 5x
* Galaxy S6
* Pixel 3
* Android API 28 emulator

**Comment by LittleAngelwings** [comment]
Can reproduce it with ftp://127
If I enter this, only a white site appears. If I press the tab/homescreen button after a short while Fenix crashes immediately.
I reproduced the steps three or four times. From this point, after the last try, Fenix will crashes immediately after restart/start. 
Only to clear apps data and apps cache will break this start crash loop and I could start Fenix. 

Reproduced with Android 8.1.0 and 8.0.0 
And Raptor 1.0.1920 (build #1) 5/16 6:32pm and nigthly 1.0.1920 (11361806)
